pontoon/base/views.py

```diff
@@ -294,19 +304,27 @@ def get_translations_from_other_locales(request):
         }, status=400)
 
     entity = get_object_or_404(Entity, pk=entity)
-    locales = entity.resource.project.locales.exclude(code=locale)
+    locale = get_object_or_404(Locale, code=locale)
+    preferred_locales = request.user.profile.sorted_locales
```

Comment:
In `models.py` we should introduce a new `preferred_locales` property and use that instead of `sorted_locales` (sorting will be lost below, so we'll have to sort again anyways):
```python
@property
def preferred_locales(self):
    locales = Locale.objects.filter(pk__in=self.locales_order)
```

We should then also make `sorted_locales` property use `preferred_locales`.